propolis/src/hw/pci/bridge.rs

```diff
@@ -110,41 +100,218 @@ impl Bridge {
     /// the bridge's secondary bus number will update the supplied router such
     /// that it maps the new bus number to the bridge's downstream bus.
     pub fn new(bus: Arc<Bus>, router: Arc<Router>) -> Arc<Self> {
-        let builder = Builder::new(super::Ident {
-            vendor_id: 0x1de,
-            prog_if: BRIDGE_PROG_IF,
-            subclass: BRIDGE_PROG_SUBCLASS,
-            class: BRIDGE_PROG_CLASS,
-            ..Default::default()
-        });
+        let cfg_builder = CfgBuilder::new();
         Arc::new(Self {
-            pci_state: builder.finish(),
+            cfg_map: cfg_builder.finish().0,
             inner: Mutex::new(Inner::new(bus, router)),
         })
     }
-}
 
-impl Device for Bridge {
-    fn device_state(&self) -> &DeviceState {
-        &self.pci_state
+    fn cfg_header_rw(&self, mut rwo: RWOp, ctx: &DispCtx) {
+        CFG_HEADER_MAP.process(&mut rwo, |id, rwo| match rwo {
+            RWOp::Read(ro) => {
+                self.cfg_std_read(id, ro, ctx);
+            }
+            RWOp::Write(wo) => {
+                self.cfg_std_write(id, wo, ctx);
+            }
+        })
     }
 
-    fn std_cfg_rw(&self, mut rwo: RWOp, ctx: &DispCtx) {
-        CFG_HEADER_MAP.process(&mut rwo, |id, rwo| match rwo {
-            RWOp::Read(ro) => match id {
-                CfgReg::Common(StdCfgReg::HeaderType) => {
-                    ro.write_u8(HEADER_TYPE_BRIDGE)
+    fn cfg_std_read(&self, id: &BridgeReg, ro: &mut ReadOp, _ctx: &DispCtx) {
+        match id {
+            BridgeReg::Common(id) => match id {
+                StdCfgReg::VendorId => ro.write_u16(BRIDGE_VENDOR_ID),
+                StdCfgReg::DeviceId => ro.write_u16(BRIDGE_DEVICE_ID),
```

Comment:
We've generally made vendor and device IDs (and their subordinate compatriots) configurable at create time. (see `PciNvme`)

Reply:
Done.